### ISSUE TITLE
Autocorrect involving heredoc puts code into heredoc

### ISSUE THREAD

**Issue by G-Rath** [rubocop/rubocop#15084]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/GuardClause

### Steps to reproduce the problem

1. create a file with the following:

```ruby
if something?
  do_something
else
  raise <<~MSG.strip
    oh noes!
  MSG
end
```

2. run `rubocop -a`

### Expected behavior

The file should contain

```ruby
raise <<~MSG.strip unless something?
  oh noes!
MSG

do_something
```

### Actual behavior

The file contains

```ruby
raise <<~MSG.strip unless something?
  do_something


    oh noes!
MSG
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.86.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.4.8) [x86_64-linux]
```

**Comment by G-Rath** [comment]
~@nzlaura just found this, you might be interested in picking it up 😄~ actually I might pick this one up

**Comment by G-Rath** [comment]
looks like this might be related to https://github.com/rubocop/rubocop/pull/14819
